### PR TITLE
Validate events

### LINKED ISSUE #793: Validate the event
**What steps did you take and what happened:**

Sent an event that did not follow the event schema.

``` TypeScript
export interface Target {
    id: string;
    name?: string;
    href?: string;
    type?: string;
    fields?: {
        [key: string]: string;
    };
}
export interface Actor {
    id: string;
    name?: string;
    href?: string;
    fields?: {
        [key: string]: string;
    };
}
export interface Group {
    id: string;
    name?: string;
}
export interface Event {
    action: string;
    group?: Group;
    crud?: string;
    created?: Date;
    actor?: Actor;
    target?: Target;
    source_ip?: string;
    description?: string;
    is_failure?: boolean;
    is_anonymous?: boolean;
    fields?: {
        [key: string]: string;
    };
}
```

**What did you expect to happen:**
The event should get processed and saved to ElasticSearch.

**Any relevant output that will help us better understand what's going on**:
In case of an event which is not according to the schema we get errors like below in `processor`
```
ResponseError: mapper_parsing_exception: 
[mapper_parsing_exception] Reason: failed to parse field [fields.sessionId] of type [float] in document with id '1678874695825-c21c233f997149269d933b734dd63fe6'.
Preview of field's value: '8.498798495498499E48'
(Pasting long output into a [GitHub gist](https://gist.github.com) or other pastebin is fine.)
```

### PR DESCRIPTION
**Pull request by ukrocks007**
Thank you for contributing to Retraced!

# Please add a summary of your change

# Does your change fix a particular issue?

Fixes #793

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,6 +21,7 @@
     "express": "4.18.2",
     "graphql": "16.6.0",
     "handlebars": "4.7.7",
+    "joi": "^17.9.1",
```
Comment (deepakprabhakara) [inline package.json:24]: pin the version

**Comment by ukrocks007** [comment]
@deepakprabhakara are we allowing teamId in the event schema?

**Review by deepakprabhakara — APPROVED** [review]

**Comment by deepakprabhakara** [comment]
> @deepakprabhakara are we allowing teamId in the event schema?

No.

File: src/handlers/createEvent.ts [hunk 2]
```diff
@@ -16,6 +16,44 @@ import Authenticator from "../security/Authenticator";
 import { logger } from "../logger";
 import config from "../config";
 
+import Joi from "joi";
+
+// Define the schema for CreateEventRequest
+const createEventRequestSchema = Joi.object({
+  action: Joi.string().required(),
+  // crud can only be one of 'c', 'r', 'u', or 'd'
+  crud: Joi.string().valid("c", "r", "u", "d").required(),
```
Comment (deepakprabhakara) [inline src/handlers/createEvent.ts:25]: crud is optional

**Review by deepakprabhakara — CHANGES_REQUESTED** [review]
crud is optional

**Review by deepakprabhakara — APPROVED** [review]

**Pull request merged**